client/payment-details/payment-method/grabpay/index.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+
+/**
+ * Internal dependencies.
+ */
+import Detail from '../detail';
+import { PaymentMethodDetails } from 'wcpay/payment-details/types';
+import { Charge } from 'wcpay/types/charges';
+
+interface GrabPayPaymentMethodDetails extends PaymentMethodDetails {
+	grabPayTransactionId: string;
+}
+
+/**
+ * Extracts and formats payment method details from a charge.
+ *
+ * @param {Charge} charge The charge object.
+ * @return {GrabPayPaymentMethodDetails} A flat hash of all necessary values.
+ */
+const formatPaymentMethodDetails = (
+	charge: Charge
+): GrabPayPaymentMethodDetails => {
+	const { billing_details: billingDetails, payment_method: id } = charge;
+	const { name, email, formatted_address: formattedAddress } = billingDetails;
+
+	const grabPayTransactionId =
+		charge.payment_method_details.grabpay?.transaction_id;
+
+	return {
+		id,
+		grabPayTransactionId,
+		name,
+		email,
+		formattedAddress,
+	};
+};
+
+interface GrabPayDetailsProps {
+	charge: Charge;
+	isLoading: boolean;
+}
+
+const GrabPayDetails: React.FC< GrabPayDetailsProps > = ( {
+	charge,
+	isLoading,
+} ) => {
+	const {
+		id,
+		grabPayTransactionId,
+		name,
+		email,
+		formattedAddress,
+	} = formatPaymentMethodDetails( charge );
+
+	return (
+		<div className="payment-method-details">
+			<div className="payment-method-details__column">
+				<Detail
+					isLoading={ isLoading }
+					label={ __( 'ID', 'woocommerce-payments' ) }
+				>
+					{ !! id ? id : '–' }
+				</Detail>
+
+				<Detail
+					isLoading={ isLoading }
+					label={ __(
+						'GrabPay Transaction ID',
+						'woocommerce-payments'
+					) }
+				>
+					{ !! grabPayTransactionId ? grabPayTransactionId : '–' }
+				</Detail>
```

Comment:
```suggestion
				<Detail
					isLoading={ isLoading }
					label={ __(
						'Transaction ID',
						'woocommerce-payments'
					) }
				>
					{ !! grabPayTransactionId ? grabPayTransactionId : '–' }
				</Detail>
```
I have two quick questions about this transaction ID. 

Firstly, I'm wondering if GrabPay offers any guidance on how to present this transaction ID. I thought about saving a little space by removing "GrabPay" from the label, but then again, this may cause confusion between our own internal transaction ID, so perhaps it is more sensible to leave this qualifier here.

Secondly, in what circumstances does the transaction ID actually appear? All of my payment intents had a `null` value for the transaction ID. Is this a GrabPay configuration or does this not appear for test mode purchases perhaps? I'm just wondering the relevance of even including this field. Interested to hear your thoughts! 🙏

Reply:
I'll remove the `Transaction ID`, since the property seems to be present in other payment methods with a null value as well (I saw the property in a `wechat_pay` transaction), so I'm assuming it won't be useful to display.